login/service_whitebox_test.go

```diff
@@ -7,142 +7,194 @@ import (
+	s.T().Log(githubEmails)
+	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), githubEmails)
+	assert.NotEmpty(s.T(), githubEmails)
+}
 
+func (s *serviceWhiteBoxTest) TestGetUser() {
+	resource.Require(s.T(), resource.UnitTest)
 	accessToken := &oauth2.Token{
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
+	githubUser, err := s.loginService.getUser(context.Background(), accessToken)
+	assert.Nil(s.T(), err)
+	assert.NotNil(s.T(), githubUser)
+	s.T().Log(githubUser)
+}
 
-	githubEmails, err := loginService.getUserEmails(context.Background(), accessToken)
-	t.Log(githubEmails)
-	assert.Nil(t, err)
-	assert.NotNil(t, githubEmails)
-	assert.NotEmpty(t, githubEmails)
+func (s *serviceWhiteBoxTest) TestFilterPrimaryEmail() {
+	resource.Require(s.T(), resource.UnitTest)
+	s.T().Skip("Not implemented")
 }
 
-func TestGetUser(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
-	setup()
-	defer tearDown()
+func (s *serviceWhiteBoxTest) TestCreateUser() {
+	resource.Require(s.T(), resource.Database)
+	defer gormsupport.DeleteCreatedEntities(s.DB)()
+	s.removeUserIfExists("sbose78@gmail.com")
 
 	accessToken := &oauth2.Token{
 		AccessToken: configuration.GetGithubAuthToken(),
 		TokenType:   "Bearer",
 	}
-
-	githubUser, err := loginService.getUser(context.Background(), accessToken)
-	assert.Nil(t, err)
-	assert.NotNil(t, githubUser)
-	t.Log(githubUser)
+	identity, err := s.loginService.getUserIdentity(context.Background(), accessToken)
+	if err != nil {
+		s.T().Log(err)
+		s.T().FailNow()
+	}
+	assert.NotNil(s.T(), identity.FullName)
+	assert.Regexp(s.T(), "https://avatars.githubusercontent.com/u/...", identity.ImageURL)
+	storedIdentity, _ := s.identityRepository.Load(context.Background(), identity.ID)
+	require.NotNil(s.T(), storedIdentity)
+	assert.NotNil(s.T(), storedIdentity.FullName)
+	assert.NotNil(s.T(), storedIdentity.ImageURL)
 }
 
-func TestFilterPrimaryEmail(t *testing.T) {
-	resource.Require(t, resource.UnitTest)
+func (s *serviceWhiteBoxTest) TestUpdateUser() {
+	resource.Require(s.T(), resource.Database)
+	defer gormsupport.DeleteCreatedEntities(s.DB)()
+	// preload db with an identity
+	s.removeUserIfExists("sbose78@gmail.com")
+	existingUser := account.User{Email: "sbose78@gmail.com"}
+	existingIdentity := account.Identity{Emails: []account.User{existingUser}}
+	s.identityRepository.Create(context.Background(), &existingIdentity)
+	// verify that the identify exists, but without any fullname and avatar URL
+	storedIdentity, _ := s.identityRepository.Load(context.Background(), existingIdentity.ID)
+	require.NotNil(s.T(), storedIdentity)
+	require.Equal(s.T(), "", storedIdentity.FullName)
+	require.Equal(s.T(), "", storedIdentity.ImageURL)
+	accessToken := &oauth2.Token{
+		AccessToken: configuration.GetGithubAuthToken(),
+		TokenType:   "Bearer",
+	}
+	identity, err := s.loginService.getUserIdentity(context.Background(), accessToken)
+	if err != nil {
```

Comment:
Just make this `require.Nil(s.T(), err)` and your test will automatically fail. No need for the if logic.